### DISCUSSION TITLE
Creating new structures in cst cell

### DISCUSSION THREAD

**Discussion by Chrismayhew321** [e0404/matRad#715] [Q&A] [answered]
I'm trying to add structures to the cst cell to place these in my beamline. I have my structure's vertices lps coordinates but converting these to ct voxel indices seems to place my objects at incorrect positions/orientations. Is there a method to convert the lps coords to ct voxel indices?

**Comment by wahln** [comment]
matRad has a very annoying multitude of coordiante systems (we are currently trying to refactor those but it's not that trivial).

First of all, you need to be wary of voxel indices (i,j,k) and coordinates (x,y,z). Matlab inherently treats images with switched coordinates (j,i,k), which is usually the reason for images becoming "transposed" when importing something manually into matRad. Is this the orientation problem you are seeing?

For the absolute coordinates, it depends on it a bit how you do it and which coordinates are stored in the ct cube. Are the vectors `ct.x`, `ct.y`, `ct.z` set for you?

**Reply by Chrismayhew321** [reply, marked as answer]
Thanks a lot, i believe I've managed to figure it out now. I was creating the structure in the bev coordinates, transposing and then converting into the ct cube - I was getting it wrong during this process.
